sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java

```diff
@@ -510,30 +518,14 @@ BigQueryIO.Write<Row> createStorageWriteApiTransform(Schema schema) {
             schema.getFieldNames().containsAll(Arrays.asList("destination", "record")),
             "When writing to dynamic destinations, we expect Row Schema with a "
                 + "\"destination\" string field and a \"record\" Row field.");
-        RowDynamicDestinations dynamicDestination =
-            new RowDynamicDestinations(schema.getField("record").getType().getRowSchema());
-
-        if (isCdcConfigured()) {
-          dynamicDestination =
-              dynamicDestination.withPrimaryKey(configuration.getUseCdcWritesWithPrimaryKey());
-          write = validateAndIncludeCDCInformation(write, schema);
-        }
         write =
             write
-                .to(dynamicDestination)
+                .to(new RowDynamicDestinations(schema.getField("record").getType().getRowSchema()))
                 .withFormatFunction(row -> BigQueryUtils.toTableRow(row.getRow("record")));
+      } else if (Optional.ofNullable(configuration.getUseCdcWrites()).orElse(false)) {
+        write = validateAndIncludeCDCInformation(write, schema);
```

Comment:
Can we make this a separate `if` block, outside of this if/else chain? We should be able to apply this method to both dynamic destination and single table cases. The only factor should be whether or not `useCdcWrites` is true